lib/iris/tests/unit/fileformats/netcdf/test_Saver__ugrid.py

```diff
@@ -0,0 +1,389 @@
+        # both the main variables reference the same mesh, and 'face' location
+        v_a, v_b = vars["a"], vars["b"]
+        self.assertEqual(v_a["mesh"], mesh_name)
+        self.assertEqual(v_a["location"], "face")
+        self.assertEqual(v_b["mesh"], mesh_name)
+        self.assertEqual(v_b["location"], "face")
+        # self.assertCDL(self.tempfile_path)
+
+    def test_multi_cubes_different_locations(self):
+        cube1 = make_cube(var_name="a", location="face")
+        cube2 = make_cube(var_name="b", location="node")
+        dims, vars = self.check_save([cube1, cube2])
+
+        # there is exactly 1 mesh in the file
+        ((mesh_name, mesh_props),) = vars_meshvars(vars).items()
+
+        # the main variables reference the same mesh at different locations
+        v_a, v_b = vars["a"], vars["b"]
+        self.assertEqual(v_a["mesh"], mesh_name)
+        self.assertEqual(v_a["location"], "face")
+        self.assertEqual(v_b["mesh"], mesh_name)
+        self.assertEqual(v_b["location"], "node")
+
+        # the main variables map the face and node dimensions
+        face_dim = vars_meshdim(vars, "face")
+        node_dim = vars_meshdim(vars, "node")
+        self.assertEqual(v_a["_dims"], [face_dim])
+        self.assertEqual(v_b["_dims"], [node_dim])
+
+    def test_multi_cubes_identical_meshes(self):
+        mesh1 = make_mesh()
+        mesh2 = make_mesh()
+        cube1 = make_cube(var_name="a", mesh=mesh1)
+        cube2 = make_cube(var_name="b", mesh=mesh2)
+        dims, vars = self.check_save([cube1, cube2])
+
+        # there are 2 meshes in the file
+        mesh_vars = vars_meshvars(vars)
+        self.assertEqual(len(mesh_vars), 2)
+
+        # there are two (data)variables with a 'mesh' property
+        mesh_datavars = vars_w_props(vars, mesh="*")
+        self.assertEqual(2, len(mesh_datavars))
+        self.assertEqual(["a", "b"], sorted(mesh_datavars.keys()))
+
+        # the data variables reference the two separate meshes
+        a_props, b_props = vars["a"], vars["b"]
+        mesh_a, mesh_b = a_props["mesh"], b_props["mesh"]
+        self.assertEqual(sorted([mesh_a, mesh_b]), sorted(mesh_vars.keys()))
+
+    def test_multi_cubes_different_mesh(self):
+        cube1 = make_cube(var_name="a")
+        cube2 = make_cube(var_name="b", mesh=make_mesh(n_faces=4))
+        self.check_save([cube1, cube2])
+        dims, vars = self.check_save([cube1, cube2])
+        # there are 2 meshes in the file
+        mesh_vars = vars_w_props(vars, cf_role="mesh_topology")
+        self.assertEqual(len(mesh_vars), 2)
+        # there are two (data)variables with a 'mesh' property
+        mesh_datavars = vars_w_props(vars, mesh="*")
+        self.assertEqual(2, len(mesh_datavars))
+        self.assertEqual(["a", "b"], sorted(mesh_datavars.keys()))
+        # the main variables reference the same mesh, and 'face' location
+        # self.assertCDL(self.tempfile_path)
```

Comment:
More checks wanted here.
Check that 'a' and 'b' have different meshes, and different compatible shapes.